src/DbPostgreSQLAdapter/Table.cpp

```diff
@@ -0,0 +1,660 @@
+		return "";
+	}
+
+	std::string getSQLValue(const systelab::db::IFieldValue& fieldValue, bool forComparison, bool forAssignment)
+	{
+		std::ostringstream fieldValueStream;
+		if (fieldValue.isNull())
+		{
+			if (forComparison)
+			{
+				fieldValueStream << " IS ";
+			}
+			else if (forAssignment)
+			{
+				fieldValueStream << " = ";
+			}
+
+			fieldValueStream << "NULL";
+		}
+		else
+		{
+			if (forComparison || forAssignment)
+			{
+				fieldValueStream << " = ";
+			}
+
+			const systelab::db::FieldTypes fieldType = fieldValue.getField().getType();
+			switch (fieldType)
+			{
+			case systelab::db::BOOLEAN:
+				fieldValueStream << (fieldValue.getBooleanValue() ? "True" : "False");
+				break;
+			case systelab::db::INT:
+				fieldValueStream << fieldValue.getIntValue();
+				break;
+			case systelab::db::DOUBLE:
+				fieldValueStream << std::fixed << std::setprecision(10) << fieldValue.getDoubleValue();
+				break;
+			case systelab::db::STRING:
+				fieldValueStream << "'" << fieldValue.getStringValue() << "'";
+				break;
+			case systelab::db::DATETIME:
+				fieldValueStream << "'" << dateTimeToISOString(fieldValue.getDateTimeValue()) << "'";
+				break;
+			case systelab::db::BINARY:
+				throw std::runtime_error("Insert of tables with binary fields not implemented.");
+				break;
+			default:
+				throw std::runtime_error("Invalid record field type.");
+				break;
+			}
+		}
+
+		return fieldValueStream.str();
+	}
+
+	std::string getStringList(const std::vector<std::string>& items, const std::string& separator)
+	{
+		std::string stringList = "";
+		if (!items.empty())
+		{
+			stringList = std::accumulate(items.begin()+1, items.end(), items.at(0),
+				[&separator](const std::string& a, const std::string& b)
+				{
+					return a + separator + b;
+				});
+		}
+
+		return stringList;
+	}
+
+	systelab::db::FieldTypes getTypeFromPostgresTypeName(std::string postgresTypeName)
+	{
+		if (postgresTypeName == "boolean")
+		{
+			return systelab::db::BOOLEAN;
+		}
+		else if (postgresTypeName == "integer")
+		{
+			return systelab::db::INT;
+		}
+		else if (postgresTypeName == "real")
+		{
+			return systelab::db::DOUBLE;
+		}
+		else if (postgresTypeName == "text" ||
+				 postgresTypeName == "character" ||
+			     postgresTypeName == "character varying")
+		{
+			return systelab::db::STRING;
+		}
+		else if (postgresTypeName == "timestamp with time zone")
+		{
+			return systelab::db::DATETIME;
+		}
+		else
+		{
+			std::string excMessage = "Postgress type name not recognized: " + postgresTypeName;
+			throw std::runtime_error(excMessage.c_str());
+		}
+	}
+}
+
+namespace systelab::db::postgresql {
+
+	Table::Table(Database& database, const std::string& name)
+		: m_database(database)
+		, m_name(name)
+	{
+		loadFields();
+		m_primaryKey = std::make_unique<PrimaryKey>(*this);
+	}
+
+	std::string Table::getName() const
+	{
+		return m_name;
+	}
+
+	const IPrimaryKey& Table::getPrimaryKey() const
+	{
+		return *(m_primaryKey.get());
```

Comment:
As commented somewhere else, you don't need to access the raw pointer (.get()) to get the reference to the inner value, just use the deference operator `*m_primaryKey`

Reply:
I'll change it